File: src/BackBundle/Entity/Produit.php
<?php

namespace BackBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;

/**
 * Produit
 *
 * @ORM\Table(name="produit")
 * @ORM\Entity(repositoryClass="BackBundle\Repository\ProduitRepository")
 * @UniqueEntity(fields={"libelleP"}, message="Ce produit est deja existant !")
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="libelle_p", type="string", length=255 , unique=true)
     */
    private $libelleP;

    public function __toString()
    {return $this->getLibelleP();
        // TODO: Implement __toString() method.
    }

    /**
     * @var float
     *
     * @ORM\Column(name="prix_achat", type="float")
     */
    private $prixAchat;

    /**
     * @var float
     *
     * @ORM\Column(name="prix_vente", type="float")
     */
    private $prixVente;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=500)
     * @Assert\File(maxSize="500k", mimeTypes={"image/jpeg", "image/jpg", "image/png", "image/GIF"})
     */
    private $image;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255)
     */
    private $description;
    /**
     * @var
     * @ORM\ManyToOne(targetEntity="BackBundle\Entity\Categorie")
     * @ORM\JoinColumn(name="id_categorie",referencedColumnName="id", onDelete="CASCADE")
     */
    private $categorie;
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_creation", type="datetime" , nullable=true)
     */
    private $dateCreation;

    /**
     * Produit constructor.
     */
    public function __construct()
    {
        $dateCreation= new Assert\DateTime();
    }

    /**
     * @return \DateTime
     */
    public function getDateCreation()
    {
        return $this->dateCreation;
    }

    /**
     * @param \DateTime $dateCreation
     */
    public function setDateCreation($dateCreation)
    {
        $this->dateCreation = $dateCreation;
    }



    /**
     * @return mixed
     */
    public function getCategorie()
    {
        return $this->categorie;
    }

    /**
     * @param mixed $etudiant
     */
    public function setCategorie($categorie)
    {
        $this->categorie = $categorie;
    }


    /**
     * @var
     * @ORM\ManyToOne(targetEntity="BackBundle\Entity\Fournisseur")
     * @ORM\JoinColumn(name="id_fournisseur",referencedColumnName="id", onDelete="CASCADE")
     */
    private $fournisseur;

    /**
     * @return mixed
     */
    public function getFournisseur()
    {
        return $this->fournisseur;
    }

    /**
     * @param mixed $fournisseur
     */
    public function setFournisseur($fournisseur)
    {
        $this->fournisseur = $fournisseur;
    }

    /**
     * @var integer
     *
     * @ORM\Column(name="nbrVues", type="integer", nullable=true)
     */
    private $nbrVues=0;

    /**
     * @return int
     */
    public function getNbrVues()
    {
        return $this->nbrVues;
    }

    /**
     * @param int $nbrVues
     */
    public function setNbrVues($nbrVues)
    {
        $this->nbrVues = $nbrVues;
    }


    /**
     * @var integer
     *
     * @ORM\Column(name="likes", type="integer", nullable=true)
     */
    private $likeCount=0;


    /**
     * @return int
     */
    public function getLikeCount()
    {
        return $this->likeCount;
    }

    /**
     * @param int $likeCount
     */
    public function setLikeCount($likeCount)
    {
        $this->likeCount = $likeCount;
    }

    /**
     * @var integer
     *
     * @ORM\Column(name="dislikes", type="integer", nullable=true)
     */
    private $dislikeCount=0;

    /**
     * @return int
     */
    public function getDislikeCount()
    {
        return $this->dislikeCount;
    }

    /**
     * @param int $dislikeCount
     */
    public function setDislikeCount($dislikeCount)
    {
        $this->dislikeCount = $dislikeCount;
    }



    /**
     * @var int
     *
     * @ORM\Column(name="rating", type="integer" , nullable=true)
     */
    private $rating=0;

    /**
     * @return int
     */
    public function getRating()
    {
        return $this->rating;
    }

    /**
     * @param int $rating
     */
    public function setRating($rating)
    {
        $this->rating = $rating;
    }


    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set libelleP
     *
     * @param string $libelleP
     *
     * @return Produit
     */
    public function setLibelleP($libelleP)
    {
        $this->libelleP = $libelleP;

        return $this;
    }

    /**
     * Get libelleP
     *
     * @return string
     */
    public function getLibelleP()
    {
        return $this->libelleP;
    }

    /**
     * Set prixAchat
     *
     * @param float $prixAchat
     *
     * @return Produit
     */
    public function setPrixAchat($prixAchat)
    {
        $this->prixAchat = $prixAchat;

        return $this;
    }

    /**
     * Get prixAchat
     *
     * @return float
     */
    public function getPrixAchat()
    {
        return $this->prixAchat;
    }

    /**
     * Set prixVente
     *
     * @param float $prixVente
     *
     * @return Produit
     */
    public function setPrixVente($prixVente)
    {
        $this->prixVente = $prixVente;

        return $this;
    }

    /**
     * Get prixVente
     *
     * @return float
     */
    public function getPrixVente()
    {
        return $this->prixVente;
    }

    /**
     * Set image
     *
     * @param string $image
     *
     * @return Produit
     */
    public function setImage($image)
    {
        $this->image = $image;

        return $this;
    }

    /**
     * Get image
     *
     * @return string
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * Set description
     *
     * @param string $description
     *
     * @return Produit
     */
    public function setDescription($description)
    {
        $this->description = $description;

        return $this;
    }

    /**
     * Get description
     *
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }
}

File: Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.maxi.myapp.entities;

import java.util.Date;

/**
 *
 * @author Lou44
 */
public class Produit {
    private int id=0;
    private Categorie id_categorie;
    private int id_fournisseur;
    private String libelle_p;
    private double prix_achat;
    private double prix_vente;
    private String image; 
    private String description;
    private Date date_creation;
    private int nbrVues;
    private int likes;
    private int dislikes;
    private int rating;

    public Produit() {
    }

    public Produit(Categorie id_categorie, String libelle_p, double prix_achat, String image, String description) {
        this.id_categorie = id_categorie;
        this.libelle_p = libelle_p;
        this.prix_achat = prix_achat;
        this.image = image;
        this.description = description;
    }
    
    public Produit(Categorie id_categorie, String libelle_p, double prix_achat, double prix_vente, String image, String description) {
        this.id_categorie = id_categorie;
        this.libelle_p = libelle_p;
        this.prix_achat = prix_achat;
        this.prix_vente = prix_vente;
        this.image = image;
        this.description = description;
    }

    public Produit(int id, Categorie id_categorie, String libelle_p, double prix_achat, double prix_vente, String image, String description) {
        this.id = id;
        this.id_categorie = id_categorie;
        this.libelle_p = libelle_p;
        this.prix_achat = prix_achat;
        this.prix_vente = prix_vente;
        this.image = image;
        this.description = description;
    }

    public Produit(Categorie id_categorie, int id_fournisseur, String libelle_p, double prix_achat, double prix_vente, String image, String description, Date date_creation, int nbrVues, int likes, int dislikes, int rating) {
        this.id_categorie = id_categorie;
        this.id_fournisseur = id_fournisseur;
        this.libelle_p = libelle_p;
        this.prix_achat = prix_achat;
        this.prix_vente = prix_vente;
        this.image = image;
        this.description = description;
        this.date_creation = date_creation;
        this.nbrVues = nbrVues;
        this.likes = likes;
        this.dislikes = dislikes;
        this.rating = rating;
    }

    public Produit(int id, Categorie id_categorie, int id_fournisseur, String libelle_p, double prix_achat, double prix_vente, String image, String description, Date date_creation, int nbrVues, int likes, int dislikes, int rating) {
        this.id = id;
        this.id_categorie = id_categorie;
        this.id_fournisseur = id_fournisseur;
        this.libelle_p = libelle_p;
        this.prix_achat = prix_achat;
        this.prix_vente = prix_vente;
        this.image = image;
        this.description = description;
        this.date_creation = date_creation;
        this.nbrVues = nbrVues;
        this.likes = likes;
        this.dislikes = dislikes;
        this.rating = rating;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Categorie getId_categorie() {
        return id_categorie;
    }

    public void setId_categorie(Categorie id_categorie) {
        this.id_categorie = id_categorie;
    }

    public int getId_fournisseur() {
        return id_fournisseur;
    }

    public void setId_fournisseur(int id_fournisseur) {
        this.id_fournisseur = id_fournisseur;
    }

    public String getLibelle_p() {
        return libelle_p;
    }

    public void setLibelle_p(String libelle_p) {
        this.libelle_p = libelle_p;
    }

    public double getPrix_achat() {
        return prix_achat;
    }

    public void setPrix_achat(double prix_achat) {
        this.prix_achat = prix_achat;
    }

    public double getPrix_vente() {
        return prix_vente;
    }

    public void setPrix_vente(double prix_vente) {
        this.prix_vente = prix_vente;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate_creation() {
        return date_creation;
    }

    public void setDate_creation(Date date_creation) {
        this.date_creation = date_creation;
    }

    public int getNbrVues() {
        return nbrVues;
    }

    public void setNbrVues(int nbrVues) {
        this.nbrVues = nbrVues;
    }

    public int getLikes() {
        return likes;
    }

    public void setLikes(int likes) {
        this.likes = likes;
    }

    public int getDislikes() {
        return dislikes;
    }

    public void setDislikes(int dislikes) {
        this.dislikes = dislikes;
    }

    public int getRating1() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    @Override
    public String toString() {
        return "Produit{" + "libelle_p=" + libelle_p + '}';
    }

    @Override
    public int hashCode() {
        int hash = 7;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Produit other = (Produit) obj;
        
        if (this.libelle_p != other.libelle_p || this.id != other.id) {
            return false;
        }
        return true;
    }

    

    
    
    
  
    

    
    
   
    
    
    
    
}
